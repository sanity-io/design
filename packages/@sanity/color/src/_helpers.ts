export function _clamp(num: number): number {
  return Math.max(Math.min(num, 255), 0)
}
