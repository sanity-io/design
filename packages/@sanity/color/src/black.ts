import {black as blackHex} from './config'
import {ColorValue} from './types'

/**
 * The black color value.
 * @public
 */
export const black: ColorValue = {title: 'Black', hex: blackHex}
