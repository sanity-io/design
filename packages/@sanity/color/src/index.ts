import * as config from './config'
import {hctToRgb} from './hctToRgb'
import {gray, red, magenta, purple, blue, cyan, green, yellow, orange} from './hues'
import {rgbToHex} from './rgbToHex'
import {ColorPalette, ColorValue} from './types'

export * from './constants'
export * from './darken'
export * from './hctToRgb'
export * from './hexToRgb'
export * from './limit'
export * from './rgbToHex'
export * from './screen'
export * from './types'

export {gray, red, magenta, purple, blue, cyan, green, yellow, orange}

export const hues = {gray, red, magenta, purple, blue, cyan, green, yellow, orange}

/**
 * The black color value.
 * @public
 */
export const black: ColorValue = {
  title: 'Black',
  hct: [config.black.hue, config.black.chroma, config.black.tone || 0],
  hex: rgbToHex(hctToRgb([config.black.hue, config.black.chroma, config.black.tone || 0])),
}

/**
 * The white color value.
 * @public
 */
export const white: ColorValue = {
  title: 'White',
  hct: [config.white.hue, config.white.chroma, config.white.tone || 100],
  hex: rgbToHex(hctToRgb([config.white.hue, config.white.chroma, config.white.tone || 100])),
}

/**
 * The Sanity color palette
 * @public
 */
export const color: ColorPalette = {
  // values
  black,
  white,

  // hues
  gray,
  red,
  magenta,
  purple,
  blue,
  cyan,
  green,
  yellow,
  orange,
}
