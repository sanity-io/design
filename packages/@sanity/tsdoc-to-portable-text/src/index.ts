export * from './extract'
export * from './load'
export * from './sanity'
export * from './transform'

export type {ExtractorMessage} from '@microsoft/api-extractor'
