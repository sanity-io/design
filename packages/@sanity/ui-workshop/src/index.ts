export type {WorkshopContextValue} from './WorkshopContext'

export * from './constants'
export * from './define'
export * from './frame'
export * from './location'
export * from './plugins'
export * from './types'
export * from './useWorkshop'
export * from './Workshop'
export * from './WorkshopProvider'
export * from './workshopReducer'
export * from './useAction'
export * from './useWorkshop'
