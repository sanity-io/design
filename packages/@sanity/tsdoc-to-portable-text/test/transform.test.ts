import {extract, transform} from '../src'
import {spawnProject} from './lib/spawnProject'

describe('transform', () => {
  test('should result in a "api.release" document', async () => {
    const project = await spawnProject('mylib/1.0.0')

    const result = await extract(project.path, {
      customTags: [{name: 'sampleCustomBlockTag', syntaxKind: 'block', allowMultiple: true}],
    })

    project.cleanup()

    const docs = transform(result, {package: {version: '1.0.0'}})

    expect(docs).toMatchSnapshot()
  })

  test('should result in "api.identifier" documents', async () => {
    const project = await spawnProject('mylib/1.0.0')

    const result = await extract(project.path, {
      customTags: [{name: 'sampleCustomBlockTag', syntaxKind: 'block', allowMultiple: true}],
    })

    project.cleanup()

    const docs = transform(result, {package: {version: '1.0.0'}})
    const identifierDocs = docs.filter((d) => d._type === 'api.identifier')

    expect(identifierDocs.length).toBe(7)
  })

  test('should transform class', async () => {
    const project = await spawnProject('mylib/1.0.0')

    const result = await extract(project.path, {
      customTags: [{name: 'sampleCustomBlockTag', syntaxKind: 'block', allowMultiple: true}],
    })

    project.cleanup()

    const docs = transform(result, {package: {version: '1.0.0'}})
    const classDoc = docs.find((d) => d._type === 'api.class')

    expect(classDoc).toMatchSnapshot()
  })

  test('should transform interface with call signature', async () => {
    const project = await spawnProject('mylib/1.0.0')

    const result = await extract(project.path, {
      customTags: [{name: 'sampleCustomBlockTag', syntaxKind: 'block', allowMultiple: true}],
    })

    project.cleanup()

    const docs = transform(result, {package: {version: '1.0.0'}})
    const interfaceDoc = docs.find((d) => d._type === 'api.interface' && d.name === 'Resolver')

    expect(interfaceDoc).toMatchSnapshot()
  })

  test('should transform multiple exports', async () => {
    const project = await spawnProject('multi-export/1.0.0')

    const result = await extract(project.path, {
      customTags: [{name: 'sampleCustomBlockTag', syntaxKind: 'block', allowMultiple: true}],
    })

    project.cleanup()

    const docs = transform(result, {package: {version: '1.0.0'}})

    const pkg = docs.find((d) => d._type === 'api.package')!

    expect(pkg.identifiers).toEqual(['extra', 'version'])

    const release = docs.find((d) => d._type === 'api.release')!

    expect(release.exports).toHaveLength(2)

    const names = (release.exports as any[])
      .reduce<any[]>((acc, x) => acc.concat(x.members), [])
      .map((x) => docs.find((d) => d._id === x._ref)!.name)

    expect(names).toEqual(['version', 'extra'])
  })
})
