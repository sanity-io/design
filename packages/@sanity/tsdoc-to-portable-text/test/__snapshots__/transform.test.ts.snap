// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform should result in a "api.release" document 1`] = `
Array [
  Object {
    "_id": "mylib",
    "_type": "api.package",
    "latestRelease": Object {
      "_ref": "mylib_1-0-0",
      "_type": "reference",
    },
    "name": "mylib",
    "scope": undefined,
  },
  Object {
    "_id": "mylib_1-0-0",
    "_type": "api.release",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
    "version": "1.0.0",
  },
  Object {
    "_id": "mylib_1-0-0__main",
    "_type": "api.export",
    "name": "mylib",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
    "path": ".",
    "release": Object {
      "_ref": "mylib_1-0-0",
      "_type": "reference",
    },
  },
  Object {
    "_id": "15cb148fb5d4f062894fe0e153b35382",
    "_type": "api.function",
    "comment": Object {
      "_type": "tsdoc.docComment",
      "customBlocks": undefined,
      "deprecated": undefined,
      "exampleBlocks": undefined,
      "modifierTags": Array [
        Object {
          "_key": "modifierTag0",
          "_type": "tsdoc.modifierTag",
          "name": "@public",
        },
      ],
      "parameters": undefined,
      "remarks": undefined,
      "returns": undefined,
      "seeBlocks": undefined,
      "summary": undefined,
    },
    "export": Object {
      "_ref": "mylib_1-0-0__main",
      "_type": "reference",
    },
    "isReactComponentType": true,
    "name": "Button",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
    "parameters": Array [
      Object {
        "_key": "param0",
        "_type": "api.parameter",
        "comment": undefined,
        "name": "props",
        "releaseTag": "public",
        "type": Array [
          Object {
            "_key": "token0",
            "_type": "api.token",
            "symbol": Object {
              "_ref": "2e98a594f4b9a7d1a4fd0f695f2dafed",
              "_type": "reference",
            },
            "text": "ButtonProps",
          },
          Object {
            "_key": "token1",
            "_type": "api.token",
            "text": " & ",
          },
          Object {
            "_key": "token2",
            "_type": "api.token",
            "text": "Omit",
          },
          Object {
            "_key": "token3",
            "_type": "api.token",
            "text": "<",
          },
          Object {
            "_key": "token4",
            "_type": "api.token",
            "text": "React.HTMLProps",
          },
          Object {
            "_key": "token5",
            "_type": "api.token",
            "text": "<",
          },
          Object {
            "_key": "token6",
            "_type": "api.token",
            "text": "HTMLButtonElement",
          },
          Object {
            "_key": "token7",
            "_type": "api.token",
            "text": ">, 'type'>",
          },
        ],
      },
    ],
    "propsType": undefined,
    "release": Object {
      "_ref": "mylib_1-0-0",
      "_type": "reference",
    },
    "releaseTag": "public",
    "returnType": Array [
      Object {
        "_key": "token0",
        "_type": "api.token",
        "text": "React.ReactElement",
      },
    ],
    "slug": Object {
      "_type": "slug",
      "current": "Button",
    },
    "typeParameters": Array [],
  },
  Object {
    "_id": "2e98a594f4b9a7d1a4fd0f695f2dafed",
    "_type": "api.interface",
    "comment": Object {
      "_type": "tsdoc.docComment",
      "customBlocks": undefined,
      "deprecated": undefined,
      "exampleBlocks": undefined,
      "modifierTags": Array [
        Object {
          "_key": "modifierTag0",
          "_type": "tsdoc.modifierTag",
          "name": "@public",
        },
      ],
      "parameters": undefined,
      "remarks": undefined,
      "returns": undefined,
      "seeBlocks": undefined,
      "summary": undefined,
    },
    "export": Object {
      "_ref": "mylib_1-0-0__main",
      "_type": "reference",
    },
    "extends": Array [
      Object {
        "_key": "extend0",
        "_type": "api.extend",
        "type": Array [
          Object {
            "_key": "token0",
            "_type": "api.token",
            "symbol": Object {
              "_ref": "03fdbeef0c01fd8e571a926d2905e46f",
              "_type": "reference",
            },
            "text": "ResponsiveMarginProps",
          },
        ],
      },
    ],
    "members": Array [
      Object {
        "_key": "member0",
        "_type": "api.propertySignature",
        "comment": undefined,
        "isOptional": true,
        "name": "tone",
        "releaseTag": "public",
        "type": Array [
          Object {
            "_key": "token0",
            "_type": "api.token",
            "symbol": Object {
              "_ref": "f8290b4913d399ece606e41443224121",
              "_type": "reference",
            },
            "text": "ButtonTone",
          },
        ],
      },
      Object {
        "_key": "member1",
        "_type": "api.propertySignature",
        "comment": undefined,
        "isOptional": true,
        "name": "type",
        "releaseTag": "public",
        "type": Array [
          Object {
            "_key": "token0",
            "_type": "api.token",
            "symbol": Object {
              "_ref": "7baa7de0358f25b593b8be235abe8fee",
              "_type": "reference",
            },
            "text": "ButtonType",
          },
        ],
      },
    ],
    "name": "ButtonProps",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
    "release": Object {
      "_ref": "mylib_1-0-0",
      "_type": "reference",
    },
    "releaseTag": "public",
    "slug": Object {
      "_type": "slug",
      "current": "ButtonProps",
    },
    "typeParameters": Array [],
  },
  Object {
    "_id": "f8290b4913d399ece606e41443224121",
    "_type": "api.typeAlias",
    "comment": Object {
      "_type": "tsdoc.docComment",
      "customBlocks": undefined,
      "deprecated": undefined,
      "exampleBlocks": undefined,
      "modifierTags": Array [
        Object {
          "_key": "modifierTag0",
          "_type": "tsdoc.modifierTag",
          "name": "@public",
        },
      ],
      "parameters": undefined,
      "remarks": undefined,
      "returns": undefined,
      "seeBlocks": undefined,
      "summary": undefined,
    },
    "export": Object {
      "_ref": "mylib_1-0-0__main",
      "_type": "reference",
    },
    "name": "ButtonTone",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
    "release": Object {
      "_ref": "mylib_1-0-0",
      "_type": "reference",
    },
    "releaseTag": "public",
    "slug": Object {
      "_type": "slug",
      "current": "ButtonTone",
    },
    "type": Array [
      Object {
        "_key": "token0",
        "_type": "api.token",
        "text": "'positive' | 'caution' | 'critical'",
      },
    ],
  },
  Object {
    "_id": "7baa7de0358f25b593b8be235abe8fee",
    "_type": "api.typeAlias",
    "comment": Object {
      "_type": "tsdoc.docComment",
      "customBlocks": undefined,
      "deprecated": undefined,
      "exampleBlocks": undefined,
      "modifierTags": Array [
        Object {
          "_key": "modifierTag0",
          "_type": "tsdoc.modifierTag",
          "name": "@public",
        },
      ],
      "parameters": undefined,
      "remarks": undefined,
      "returns": undefined,
      "seeBlocks": undefined,
      "summary": undefined,
    },
    "export": Object {
      "_ref": "mylib_1-0-0__main",
      "_type": "reference",
    },
    "name": "ButtonType",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
    "release": Object {
      "_ref": "mylib_1-0-0",
      "_type": "reference",
    },
    "releaseTag": "public",
    "slug": Object {
      "_type": "slug",
      "current": "ButtonType",
    },
    "type": Array [
      Object {
        "_key": "token0",
        "_type": "api.token",
        "text": "'button' | 'submit' | 'reset'",
      },
    ],
  },
  Object {
    "_id": "ead4d7d99c69e54e6e02479a21215ad0",
    "_type": "api.class",
    "comment": Object {
      "_type": "tsdoc.docComment",
      "customBlocks": Array [
        Object {
          "_key": "customBlock1",
          "_type": "tsdoc.customBlock",
          "content": Array [
            Object {
              "_key": "node0",
              "_type": "block",
              "children": Array [
                Object {
                  "_key": "node0",
                  "_type": "span",
                  "marks": Array [],
                  "text": "You can define your own custom block tags and tell the parser about them. The playground doesn't render this block, but your own tooling could do that.",
                },
              ],
              "markDefs": Array [],
              "style": "normal",
            },
          ],
          "tag": "@sampleCustomBlockTag",
        },
      ],
      "deprecated": undefined,
      "exampleBlocks": Array [
        Object {
          "_key": "exampleBlock0",
          "_type": "tsdoc.exampleBlock",
          "content": Array [
            Object {
              "_key": "node0",
              "_type": "block",
              "children": Array [
                Object {
                  "_key": "node0",
                  "_type": "span",
                  "marks": Array [],
                  "text": "Code samples can be enclosed in a Markdown code fence:",
                },
              ],
              "markDefs": Array [],
              "style": "normal",
            },
            Object {
              "_key": "node1",
              "_type": "code",
              "code": "function getAverage(x, y) {
  return (x + y) / 2.0;
}
",
              "language": "ts",
            },
          ],
        },
      ],
      "modifierTags": Array [
        Object {
          "_key": "modifierTag0",
          "_type": "tsdoc.modifierTag",
          "name": "@beta",
        },
        Object {
          "_key": "modifierTag1",
          "_type": "tsdoc.modifierTag",
          "name": "@sealed",
        },
      ],
      "parameters": undefined,
      "remarks": Object {
        "_type": "tsdoc.remarksBlock",
        "content": Array [
          Object {
            "_key": "node0",
            "_type": "block",
            "children": Array [
              Object {
                "_key": "node0",
                "_type": "span",
                "marks": Array [],
                "text": "The main documentation for an API item is separated into a brief \\"summary\\" section optionally followed by an ",
              },
              Object {
                "_key": "node1",
                "_type": "span",
                "marks": Array [
                  "code",
                ],
                "text": "@remarks",
              },
              Object {
                "_key": "node2",
                "_type": "span",
                "marks": Array [],
                "text": " block containing additional details.",
              },
            ],
            "markDefs": Array [],
            "style": "normal",
          },
          Object {
            "_key": "node1",
            "_type": "block",
            "children": Array [
              Object {
                "_key": "node0",
                "_type": "span",
                "marks": Array [],
                "text": "Unlike the summary, the remarks block may contain lengthy documentation content. The remarks should not restate information from the summary, since the summary section will always be displayed wherever the remarks section appears. Other sections (e.g. an ",
              },
              Object {
                "_key": "node1",
                "_type": "span",
                "marks": Array [
                  "code",
                ],
                "text": "@example",
              },
              Object {
                "_key": "node2",
                "_type": "span",
                "marks": Array [],
                "text": " block) if shown will usually appear after the remarks section.",
              },
            ],
            "markDefs": Array [],
            "style": "normal",
          },
          Object {
            "_key": "node2",
            "_type": "block",
            "children": Array [
              Object {
                "_key": "node0",
                "_type": "span",
                "marks": Array [],
                "text": "Code samples can be enclosed in a Markdown code fence:",
              },
            ],
            "markDefs": Array [],
            "style": "normal",
          },
          Object {
            "_key": "node3",
            "_type": "code",
            "code": "function getAverage(x, y) {
  return (x + y) / 2.0;
}
",
            "language": "ts",
          },
          Object {
            "_key": "node4",
            "_type": "block",
            "children": Array [
              Object {
                "_key": "node0",
                "_type": "span",
                "marks": Array [],
                "text": "If you use special symbols and don't want to use a code span, you can escape them using backslashes: ",
              },
              Object {
                "_key": "node1",
                "_type": "span",
                "marks": Array [],
                "text": "{",
              },
              Object {
                "_key": "node2",
                "_type": "span",
                "marks": Array [],
                "text": "@",
              },
              Object {
                "_key": "node3",
                "_type": "span",
                "marks": Array [],
                "text": "inheritDoc Button.render",
              },
              Object {
                "_key": "node4",
                "_type": "span",
                "marks": Array [],
                "text": "}",
              },
            ],
            "markDefs": Array [],
            "style": "normal",
          },
        ],
      },
      "returns": undefined,
      "seeBlocks": undefined,
      "summary": Array [
        Object {
          "_key": "node0",
          "_type": "block",
          "children": Array [
            Object {
              "_key": "node0",
              "_type": "span",
              "marks": Array [],
              "text": "The summary section should be brief. On a documentation web site, it will be shown on a page that lists summaries for many different API items. On a detail page for a single item, the summary will be shown followed by the remarks section (if any).",
            },
          ],
          "markDefs": Array [],
          "style": "normal",
        },
      ],
    },
    "export": Object {
      "_ref": "mylib_1-0-0__main",
      "_type": "reference",
    },
    "members": Array [
      Object {
        "_key": "member0",
        "_type": "api.constructor",
        "comment": Object {
          "_type": "tsdoc.docComment",
          "customBlocks": undefined,
          "deprecated": undefined,
          "exampleBlocks": undefined,
          "modifierTags": undefined,
          "parameters": undefined,
          "remarks": undefined,
          "returns": undefined,
          "seeBlocks": undefined,
          "summary": Array [
            Object {
              "_key": "node0",
              "_type": "block",
              "children": Array [
                Object {
                  "_key": "node0",
                  "_type": "span",
                  "marks": Array [],
                  "text": "Constructs a new instance of the ",
                },
                Object {
                  "_key": "node1",
                  "_type": "span",
                  "marks": Array [
                    "code",
                  ],
                  "text": "Class",
                },
                Object {
                  "_key": "node2",
                  "_type": "span",
                  "marks": Array [],
                  "text": " class",
                },
              ],
              "markDefs": Array [],
              "style": "normal",
            },
          ],
        },
        "parameters": Array [
          Object {
            "_key": "param0",
            "_type": "api.parameter",
            "comment": undefined,
            "name": "id",
            "releaseTag": "beta",
            "type": Array [
              Object {
                "_key": "token0",
                "_type": "api.token",
                "text": "string",
              },
            ],
          },
        ],
        "releaseTag": "beta",
      },
      Object {
        "_key": "member1",
        "_type": "api.property",
        "comment": undefined,
        "isEventProperty": false,
        "isOptional": false,
        "isStatic": false,
        "name": "$id",
        "releaseTag": "beta",
        "type": Array [
          Object {
            "_key": "token0",
            "_type": "api.token",
            "text": "string",
          },
        ],
      },
      Object {
        "_key": "member2",
        "_type": "api.method",
        "comment": undefined,
        "isOptional": false,
        "isStatic": false,
        "name": "method",
        "parameters": Array [
          Object {
            "_key": "param0",
            "_type": "api.parameter",
            "comment": undefined,
            "name": "flag",
            "releaseTag": "beta",
            "type": Array [
              Object {
                "_key": "token0",
                "_type": "api.token",
                "text": "boolean",
              },
            ],
          },
        ],
        "releaseTag": "beta",
        "returnType": Array [
          Object {
            "_key": "token0",
            "_type": "api.token",
            "text": "boolean",
          },
        ],
        "typeParameters": Array [],
      },
    ],
    "name": "Class",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
    "release": Object {
      "_ref": "mylib_1-0-0",
      "_type": "reference",
    },
    "releaseTag": "beta",
    "slug": Object {
      "_type": "slug",
      "current": "Class",
    },
  },
  Object {
    "_id": "9053f9d2ff132c36903fbaf0067d5ce8",
    "_type": "api.interface",
    "comment": Object {
      "_type": "tsdoc.docComment",
      "customBlocks": undefined,
      "deprecated": undefined,
      "exampleBlocks": undefined,
      "modifierTags": Array [
        Object {
          "_key": "modifierTag0",
          "_type": "tsdoc.modifierTag",
          "name": "@beta",
        },
      ],
      "parameters": undefined,
      "remarks": undefined,
      "returns": undefined,
      "seeBlocks": undefined,
      "summary": undefined,
    },
    "export": Object {
      "_ref": "mylib_1-0-0__main",
      "_type": "reference",
    },
    "extends": Array [],
    "members": Array [
      Object {
        "_key": "member0",
        "_type": "api.callSignature",
        "comment": undefined,
        "members": Array [],
        "parameters": Array [
          Object {
            "_key": "param0",
            "_type": "api.parameter",
            "comment": undefined,
            "name": "id",
            "releaseTag": "beta",
            "type": Array [
              Object {
                "_key": "token0",
                "_type": "api.token",
                "text": "string",
              },
            ],
          },
        ],
        "releaseTag": "beta",
        "returnType": Array [
          Object {
            "_key": "token0",
            "_type": "api.token",
            "text": "string[]",
          },
        ],
        "typeParameters": Array [],
      },
    ],
    "name": "Resolver",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
    "release": Object {
      "_ref": "mylib_1-0-0",
      "_type": "reference",
    },
    "releaseTag": "beta",
    "slug": Object {
      "_type": "slug",
      "current": "Resolver",
    },
    "typeParameters": Array [],
  },
  Object {
    "_id": "03fdbeef0c01fd8e571a926d2905e46f",
    "_type": "api.interface",
    "comment": Object {
      "_type": "tsdoc.docComment",
      "customBlocks": undefined,
      "deprecated": undefined,
      "exampleBlocks": undefined,
      "modifierTags": Array [
        Object {
          "_key": "modifierTag0",
          "_type": "tsdoc.modifierTag",
          "name": "@public",
        },
      ],
      "parameters": undefined,
      "remarks": undefined,
      "returns": undefined,
      "seeBlocks": undefined,
      "summary": undefined,
    },
    "export": Object {
      "_ref": "mylib_1-0-0__main",
      "_type": "reference",
    },
    "extends": Array [],
    "members": Array [
      Object {
        "_key": "member0",
        "_type": "api.propertySignature",
        "comment": undefined,
        "isOptional": true,
        "name": "margin",
        "releaseTag": "public",
        "type": Array [
          Object {
            "_key": "token0",
            "_type": "api.token",
            "text": "number | number[]",
          },
        ],
      },
    ],
    "name": "ResponsiveMarginProps",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
    "release": Object {
      "_ref": "mylib_1-0-0",
      "_type": "reference",
    },
    "releaseTag": "public",
    "slug": Object {
      "_type": "slug",
      "current": "ResponsiveMarginProps",
    },
    "typeParameters": Array [],
  },
  Object {
    "_id": "mylib_Button",
    "_type": "api.symbol",
    "name": "Button",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
  },
  Object {
    "_id": "mylib_ButtonProps",
    "_type": "api.symbol",
    "name": "ButtonProps",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
  },
  Object {
    "_id": "mylib_ButtonTone",
    "_type": "api.symbol",
    "name": "ButtonTone",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
  },
  Object {
    "_id": "mylib_ButtonType",
    "_type": "api.symbol",
    "name": "ButtonType",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
  },
  Object {
    "_id": "mylib_Class",
    "_type": "api.symbol",
    "name": "Class",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
  },
  Object {
    "_id": "mylib_Resolver",
    "_type": "api.symbol",
    "name": "Resolver",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
  },
  Object {
    "_id": "mylib_ResponsiveMarginProps",
    "_type": "api.symbol",
    "name": "ResponsiveMarginProps",
    "package": Object {
      "_ref": "mylib",
      "_type": "reference",
    },
  },
]
`;

exports[`transform should transform class 1`] = `
Object {
  "_id": "ead4d7d99c69e54e6e02479a21215ad0",
  "_type": "api.class",
  "comment": Object {
    "_type": "tsdoc.docComment",
    "customBlocks": Array [
      Object {
        "_key": "customBlock1",
        "_type": "tsdoc.customBlock",
        "content": Array [
          Object {
            "_key": "node0",
            "_type": "block",
            "children": Array [
              Object {
                "_key": "node0",
                "_type": "span",
                "marks": Array [],
                "text": "You can define your own custom block tags and tell the parser about them. The playground doesn't render this block, but your own tooling could do that.",
              },
            ],
            "markDefs": Array [],
            "style": "normal",
          },
        ],
        "tag": "@sampleCustomBlockTag",
      },
    ],
    "deprecated": undefined,
    "exampleBlocks": Array [
      Object {
        "_key": "exampleBlock0",
        "_type": "tsdoc.exampleBlock",
        "content": Array [
          Object {
            "_key": "node0",
            "_type": "block",
            "children": Array [
              Object {
                "_key": "node0",
                "_type": "span",
                "marks": Array [],
                "text": "Code samples can be enclosed in a Markdown code fence:",
              },
            ],
            "markDefs": Array [],
            "style": "normal",
          },
          Object {
            "_key": "node1",
            "_type": "code",
            "code": "function getAverage(x, y) {
  return (x + y) / 2.0;
}
",
            "language": "ts",
          },
        ],
      },
    ],
    "modifierTags": Array [
      Object {
        "_key": "modifierTag0",
        "_type": "tsdoc.modifierTag",
        "name": "@beta",
      },
      Object {
        "_key": "modifierTag1",
        "_type": "tsdoc.modifierTag",
        "name": "@sealed",
      },
    ],
    "parameters": undefined,
    "remarks": Object {
      "_type": "tsdoc.remarksBlock",
      "content": Array [
        Object {
          "_key": "node0",
          "_type": "block",
          "children": Array [
            Object {
              "_key": "node0",
              "_type": "span",
              "marks": Array [],
              "text": "The main documentation for an API item is separated into a brief \\"summary\\" section optionally followed by an ",
            },
            Object {
              "_key": "node1",
              "_type": "span",
              "marks": Array [
                "code",
              ],
              "text": "@remarks",
            },
            Object {
              "_key": "node2",
              "_type": "span",
              "marks": Array [],
              "text": " block containing additional details.",
            },
          ],
          "markDefs": Array [],
          "style": "normal",
        },
        Object {
          "_key": "node1",
          "_type": "block",
          "children": Array [
            Object {
              "_key": "node0",
              "_type": "span",
              "marks": Array [],
              "text": "Unlike the summary, the remarks block may contain lengthy documentation content. The remarks should not restate information from the summary, since the summary section will always be displayed wherever the remarks section appears. Other sections (e.g. an ",
            },
            Object {
              "_key": "node1",
              "_type": "span",
              "marks": Array [
                "code",
              ],
              "text": "@example",
            },
            Object {
              "_key": "node2",
              "_type": "span",
              "marks": Array [],
              "text": " block) if shown will usually appear after the remarks section.",
            },
          ],
          "markDefs": Array [],
          "style": "normal",
        },
        Object {
          "_key": "node2",
          "_type": "block",
          "children": Array [
            Object {
              "_key": "node0",
              "_type": "span",
              "marks": Array [],
              "text": "Code samples can be enclosed in a Markdown code fence:",
            },
          ],
          "markDefs": Array [],
          "style": "normal",
        },
        Object {
          "_key": "node3",
          "_type": "code",
          "code": "function getAverage(x, y) {
  return (x + y) / 2.0;
}
",
          "language": "ts",
        },
        Object {
          "_key": "node4",
          "_type": "block",
          "children": Array [
            Object {
              "_key": "node0",
              "_type": "span",
              "marks": Array [],
              "text": "If you use special symbols and don't want to use a code span, you can escape them using backslashes: ",
            },
            Object {
              "_key": "node1",
              "_type": "span",
              "marks": Array [],
              "text": "{",
            },
            Object {
              "_key": "node2",
              "_type": "span",
              "marks": Array [],
              "text": "@",
            },
            Object {
              "_key": "node3",
              "_type": "span",
              "marks": Array [],
              "text": "inheritDoc Button.render",
            },
            Object {
              "_key": "node4",
              "_type": "span",
              "marks": Array [],
              "text": "}",
            },
          ],
          "markDefs": Array [],
          "style": "normal",
        },
      ],
    },
    "returns": undefined,
    "seeBlocks": undefined,
    "summary": Array [
      Object {
        "_key": "node0",
        "_type": "block",
        "children": Array [
          Object {
            "_key": "node0",
            "_type": "span",
            "marks": Array [],
            "text": "The summary section should be brief. On a documentation web site, it will be shown on a page that lists summaries for many different API items. On a detail page for a single item, the summary will be shown followed by the remarks section (if any).",
          },
        ],
        "markDefs": Array [],
        "style": "normal",
      },
    ],
  },
  "export": Object {
    "_ref": "mylib_1-0-0__main",
    "_type": "reference",
  },
  "members": Array [
    Object {
      "_key": "member0",
      "_type": "api.constructor",
      "comment": Object {
        "_type": "tsdoc.docComment",
        "customBlocks": undefined,
        "deprecated": undefined,
        "exampleBlocks": undefined,
        "modifierTags": undefined,
        "parameters": undefined,
        "remarks": undefined,
        "returns": undefined,
        "seeBlocks": undefined,
        "summary": Array [
          Object {
            "_key": "node0",
            "_type": "block",
            "children": Array [
              Object {
                "_key": "node0",
                "_type": "span",
                "marks": Array [],
                "text": "Constructs a new instance of the ",
              },
              Object {
                "_key": "node1",
                "_type": "span",
                "marks": Array [
                  "code",
                ],
                "text": "Class",
              },
              Object {
                "_key": "node2",
                "_type": "span",
                "marks": Array [],
                "text": " class",
              },
            ],
            "markDefs": Array [],
            "style": "normal",
          },
        ],
      },
      "parameters": Array [
        Object {
          "_key": "param0",
          "_type": "api.parameter",
          "comment": undefined,
          "name": "id",
          "releaseTag": "beta",
          "type": Array [
            Object {
              "_key": "token0",
              "_type": "api.token",
              "text": "string",
            },
          ],
        },
      ],
      "releaseTag": "beta",
    },
    Object {
      "_key": "member1",
      "_type": "api.property",
      "comment": undefined,
      "isEventProperty": false,
      "isOptional": false,
      "isStatic": false,
      "name": "$id",
      "releaseTag": "beta",
      "type": Array [
        Object {
          "_key": "token0",
          "_type": "api.token",
          "text": "string",
        },
      ],
    },
    Object {
      "_key": "member2",
      "_type": "api.method",
      "comment": undefined,
      "isOptional": false,
      "isStatic": false,
      "name": "method",
      "parameters": Array [
        Object {
          "_key": "param0",
          "_type": "api.parameter",
          "comment": undefined,
          "name": "flag",
          "releaseTag": "beta",
          "type": Array [
            Object {
              "_key": "token0",
              "_type": "api.token",
              "text": "boolean",
            },
          ],
        },
      ],
      "releaseTag": "beta",
      "returnType": Array [
        Object {
          "_key": "token0",
          "_type": "api.token",
          "text": "boolean",
        },
      ],
      "typeParameters": Array [],
    },
  ],
  "name": "Class",
  "package": Object {
    "_ref": "mylib",
    "_type": "reference",
  },
  "release": Object {
    "_ref": "mylib_1-0-0",
    "_type": "reference",
  },
  "releaseTag": "beta",
  "slug": Object {
    "_type": "slug",
    "current": "Class",
  },
}
`;

exports[`transform should transform interface with call signature 1`] = `
Object {
  "_id": "9053f9d2ff132c36903fbaf0067d5ce8",
  "_type": "api.interface",
  "comment": Object {
    "_type": "tsdoc.docComment",
    "customBlocks": undefined,
    "deprecated": undefined,
    "exampleBlocks": undefined,
    "modifierTags": Array [
      Object {
        "_key": "modifierTag0",
        "_type": "tsdoc.modifierTag",
        "name": "@beta",
      },
    ],
    "parameters": undefined,
    "remarks": undefined,
    "returns": undefined,
    "seeBlocks": undefined,
    "summary": undefined,
  },
  "export": Object {
    "_ref": "mylib_1-0-0__main",
    "_type": "reference",
  },
  "extends": Array [],
  "members": Array [
    Object {
      "_key": "member0",
      "_type": "api.callSignature",
      "comment": undefined,
      "members": Array [],
      "parameters": Array [
        Object {
          "_key": "param0",
          "_type": "api.parameter",
          "comment": undefined,
          "name": "id",
          "releaseTag": "beta",
          "type": Array [
            Object {
              "_key": "token0",
              "_type": "api.token",
              "text": "string",
            },
          ],
        },
      ],
      "releaseTag": "beta",
      "returnType": Array [
        Object {
          "_key": "token0",
          "_type": "api.token",
          "text": "string[]",
        },
      ],
      "typeParameters": Array [],
    },
  ],
  "name": "Resolver",
  "package": Object {
    "_ref": "mylib",
    "_type": "reference",
  },
  "release": Object {
    "_ref": "mylib_1-0-0",
    "_type": "reference",
  },
  "releaseTag": "beta",
  "slug": Object {
    "_type": "slug",
    "current": "Resolver",
  },
  "typeParameters": Array [],
}
`;
